ReactiveCocoa/Swift/Flatten.swift

```diff
@@ -107,6 +260,53 @@ extension SignalProducerType where Value: SignalType, Error == Value.Error {
 	}
 }
 
+extension SignalProducerType where Value: SignalType, Error == NoError {
+	/// Flattens the inner signals sent upon `producer` (into a single producer of
+	/// values), according to the semantics of the given strategy.
+	///
+	/// If an active inner signal emits an error, the returned producer will
+	/// forward that error immediately.
+	///
+	/// `Interrupted` events on inner signals will be treated like `Completed`
+	/// events on inner signals.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func flatten(strategy: FlattenStrategy) -> SignalProducer<Value.Value, Value.Error> {
+		return self
+			.map(SignalProducer.init)
+			.promoteErrors(Value.Error.self)
+			.flatten(strategy)
+	}
+}
+
+extension SignalProducerType where Value: SignalType, Error == NoError, Value.Error == NoError {
+	/// Flattens the inner signals sent upon `producer` (into a single producer of
+	/// values), according to the semantics of the given strategy.
+	///
+	/// `Interrupted` events on inner signals will be treated like `Completed`
+	/// events on inner signals.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func flatten(strategy: FlattenStrategy) -> SignalProducer<Value.Value, Value.Error> {
+		return self.map(SignalProducer.init).flatten(strategy)
+	}
+}
+
+extension SignalProducerType where Value: SignalType, Value.Error == NoError {
+	/// Flattens the inner signals sent upon `producer` (into a single producer of
+	/// values), according to the semantics of the given strategy.
+	///
+	/// If `producer` emits an error, the returned producer will forward that
+	/// error immediately.
+	///
+	/// `Interrupted` events on inner signals will be treated like `Completed`
+	/// events on inner signals.
+	@warn_unused_result(message="Did you forget to call `observe` on the signal?")
+	public func flatten(strategy: FlattenStrategy) -> SignalProducer<Value.Value, Error> {
+		return self
+			.map(SignalProducer.init)
+			.flatMap(strategy) { $0.promoteErrors(Error.self) }
```

Comment:
This could be written as `return self.flatMap(strategy) { $0.promoteErrors(Error.self) }` by [this](https://github.com/ReactiveCocoa/ReactiveCocoa/pull/2885/files#diff-b06d9f2cdc22697e0202bb4fa92ca00dR795).